apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -1503,4 +1828,156 @@ private void writeDebugLogMessages() {
+      String fieldName,
+      RifColumnType type,
+      Optional<Integer> columnLength,
+      Optional<Integer> columnScale) {
+    LinkedList<AnnotationSpec> annotations = new LinkedList<>();
+
+    // Add an @Id annotation, if appropriate.
+    if (isId) {
+      AnnotationSpec.Builder idAnnotation = AnnotationSpec.builder(Id.class);
+      annotations.add(idAnnotation.build());
+    }
+    // Add an @Column annotation to every non-transient column.
+    if (!isTransient) {
+      AnnotationSpec.Builder columnAnnotation =
+          AnnotationSpec.builder(Column.class)
+              .addMember("name", "$S", "`" + fieldName + "`")
+              .addMember("nullable", "$L", isColumnOptional);
+      if (type == RifColumnType.CHAR && columnLength.isPresent()) {
+        columnAnnotation.addMember("length", "$L", columnLength.get());
+      } else if (type == RifColumnType.NUM) {
+        /*
+         * In SQL, the precision is the number of digits in the unscaled value, e.g.
+         * "123.45" has a precision of 5. The scale is the number of digits to the right
+         * of the decimal point, e.g. "123.45" has a scale of 2.
+         */
+
+        if (columnLength.isPresent() && columnScale.isPresent()) {
+          columnAnnotation.addMember("precision", "$L", columnLength.get());
+          columnAnnotation.addMember("scale", "$L", columnScale.get());
+        } else {
+          /*
+           * Unfortunately, Hibernate's SQL schema generation (HBM2DDL) doesn't correctly
+           * handle SQL numeric datatypes that don't have a defined precision and scale.
+           * What it _should_ do is represent those types in PostgreSQL as a "NUMERIC",
+           * but what it does instead is insert a default precision and scale as
+           * "NUMBER(19, 2)". The only way to force the correct behavior is to specify a
+           * columnDefinition, so we do that. This leads to incorrect behavior with HSQL
+           * (for different reasons), but fortunately that doesn't happen to cause
+           * problems with our tests.
+           */
+          StringBuilder columnDefinition = new StringBuilder();
+          columnDefinition.append("numeric");
+          if (columnLength.isPresent() || columnScale.isPresent()) {
+            columnDefinition.append('(');
+            if (columnLength.isPresent()) {
+              columnDefinition.append(columnLength.get());
+            }
+            if (columnScale.isPresent()) {
+              columnDefinition.append(", ");
+              columnDefinition.append(columnScale.get());
+            }
+            columnDefinition.append(')');
+          }
+          columnAnnotation.addMember("columnDefinition", "$S", columnDefinition.toString());
+        }
+      }
+      annotations.add(columnAnnotation.build());
+    } else {
+      annotations.add(AnnotationSpec.builder(Transient.class).build());
+    }
+
+    return annotations;
+  }
+
+  private static TypeName selectJavaFieldType(
```

Comment:
FWIW, I'd suggest JavaDoc'ing this, as it's not entirely obvious what some of those parameters mean (I can never quite remember what scale means, for instance).

Reply:
added